pyface/ui/qt4/fields/field.py

```diff
@@ -0,0 +1,94 @@
+#------------------------------------------------------------------------------
+# Copyright (c) 2017-19, Enthought, Inc.
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+#
+# Author: Enthought, Inc.
+# Description: <Enthought pyface package component>
+#------------------------------------------------------------------------------
+""" The Qt-specific implementation of the text field class """
+
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from traits.api import Any, Instance, Unicode, provides
+
+from pyface.qt.QtCore import Qt
+from pyface.fields.i_field import IField, MField
+from pyface.ui.qt4.widget import Widget
+
+
+@provides(IField)
+class Field(MField, Widget):
+    """ The Qt-specific implementation of the field class
+
+    This is an abstract class which is not meant to be instantiated.
+    """
+
+    #: The value held by the field.
+    value = Any
+
+    #: A tooltip for the field.
+    tooltip = Unicode
+
+    #: An optional context menu for the field.
+    context_menu = Instance('pyface.action.menu_manager.MenuManager')
+
+    # ------------------------------------------------------------------------
+    # IField interface
+    # ------------------------------------------------------------------------
+
+    def _initialize_control(self, control):
+        """ Perform any toolkit-specific initialization for the control. """
+        control.setToolTip(self.tooltip)
+        control.setEnabled(self.enabled)
+        control.setVisible(self.visible)
+
+    # ------------------------------------------------------------------------
+    # IWidget interface
+    # ------------------------------------------------------------------------
+
+    def _add_event_listeners(self):
+        """ Set up toolkit-specific bindings for events """
+        super(Field, self)._add_event_listeners()
+        if self.control is not None:
+            self.control.customContextMenuRequested.connect(
+                    self._handle_context_menu)
+            self.control.contextMenuPolicy(Qt.CustomContextMenu)
```

Comment:
This is different from the behavior down below in `_context_menu_changed`. Was that intentional?